presidio-analyzer/install_nlp_models.py

```diff
@@ -0,0 +1,58 @@
+"""Install the default NLP models defined in the provided yaml file."""
+
+import logging
+import argparse
+import spacy
+import yaml
+
+try:
+    import stanza
+except ImportError:
+    # stanza should be installed manually
+    stanza = None
+
+logger = logging.getLogger()
+logger.setLevel("INFO")
+logger.addHandler(logging.StreamHandler())
+
+
+def install_models(conf_file: str) -> None:
+    """Installs models in conf/default.yaml."""
+    nlp_configuration = yaml.safe_load(open(conf_file))
+
+    logger.info(f"Installing models from configuration: {nlp_configuration}")
+
+    if "nlp_engine_name" not in nlp_configuration:
+        raise ValueError("NLP config file should contain an nlp_engine_name field")
+
+    if "models" not in nlp_configuration:
+        raise ValueError("NLP config file should contain a list of models")
+
+    for model in nlp_configuration["models"]:
+        if nlp_configuration["nlp_engine_name"] == "spacy":
+            spacy.cli.download(model["model_name"])
+        elif nlp_configuration["nlp_engine_name"] == "stanza":
+            if not stanza:
+                raise ValueError("the Stanza package should be installed manually")
+            else:
+                stanza.download(model["model_name"])
+        else:
+            raise ValueError(
+                f"Unsupported nlp engine: {nlp_configuration['nlp_engine_name']}"
+            )
+    logger.info("finished installing models")
+
```

Comment:
The if else here is extremely confusing. how about a switch case?

Reply:
I agree this is confusing, but there is no switch case in Python. How would you suggest to do it?

Reply:
Let me refactor a little bit and see if you find it more readable.

Reply:
how about using and array or lambda, somthing like (this will not work, just quick pseudo code)
```
nlp_modles = {
"spacy" : lambda s:  spacy.cli.download(model["model_name"]),
"stanza": lambda s:  stanza.download(model["model_name"])
}

nlp_models.get(nlp_configuration["nlp_engine_name"])
```

Reply:
Thanks @navalev, this is a great suggestion. I did something a little different before seeing your suggestion. If you still feel it's cumbersome I could definitely refactor again.

Reply:
@shiranr could you please review if the new code makes more sense?

Reply:
@omri374 looks good to me :)